online_party_system/party_match.go

```diff
@@ -93,24 +93,86 @@ type PartyMatchState struct {
 	joinRequests      map[string]runtime.Presence  // User IDs that have requested to join the party.
 	joinMetadata      map[string]map[string]string // User IDs to metadata mapping.
 
+	blockedUsers      map[string]struct{} // Set of users that are blocked from joining (this set is created from users blocked friends as they are invited)
+	blockedUsersDirty bool                // Do we need to re-build the blocked-list (happens when user leaves)
+
 	label             *PartyMatchLabel // Label exposed to Nakama's match listing system.
 	creator           string           // Party creator user ID, used to allow the user to automatically join when they create.
 	initialEmptyTicks int              // Number of ticks the party has been empty on creation, to ensure parties are cleaned up if their creator never joins.
 }
 
 type PartyConfig struct {
-	MaxSize        			int           			// Maximum number of members allowed in party
-	InviteDuration 			time.Duration 			// Duration an invite should last before expiring. default 0 = no expiration
-	MatchJoinMetadataFilter MatchJoinMetadataFilter	// 
-	MatchEndHook            MatchEndHook			// Called when the party match is over
-	MatchInitHook           MatchInitHook			// Called when the party match is initialized
-	MatchJoinAttemptHook    MatchJoinAttemptHook	// Called when a user attempts to join the party
-	MatchKickHook 			MatchKickHook			// Called when a party member is kicked from the party
-	MatchLeaveHook 			MatchLeaveHook			// Called when a party member leaves the party
+	MaxSize                 int                     // Maximum number of members allowed in party
+	InviteDuration          time.Duration           // Duration an invite should last before expiring. default 0 = no expiration
+	MatchJoinMetadataFilter MatchJoinMetadataFilter //
+	MatchEndHook            MatchEndHook            // Called when the party match is over
+	MatchInitHook           MatchInitHook           // Called when the party match is initialized
+	MatchJoinAttemptHook    MatchJoinAttemptHook    // Called when a user attempts to join the party
+	MatchKickHook           MatchKickHook           // Called when a party member is kicked from the party
+	MatchLeaveHook          MatchLeaveHook          // Called when a party member leaves the party
 }
 
 type PartyMatch struct {
-	config                  PartyConfig
+	config PartyConfig
+}
+
+func addUsersBlockedFriendsToBlockedSet(ctx context.Context, logger runtime.Logger, nk runtime.NakamaModule, userID string, blockedUsers map[string]struct{}) {
```

Comment:
As a general pattern I'd prefer we return the blocked users rather than mutate a given structure.

Reply:
yup